src/xo-app/settings/ips/index.js

```diff
@@ -0,0 +1,304 @@
+    const addresses = {}
+    forEach(parseIpPattern(this.state.newIp[id]), ip => {
+      addresses[ip] = {}
+    })
+    setIpPool({
+      id,
+      addresses
+    })
+    this.setState({ newIp: { ...this.state.newIp, [id]: '' } })
+  }
+  _deleteIp = ({ id, ip }) => {
+    const toBeRemoved = {}
+    if (isObject(ip)) {
+      let currentIp = ip.first
+      while (currentIp !== ip.last) {
+        toBeRemoved[currentIp] = null
+        currentIp = getNextIpV4(currentIp)
+      }
+      toBeRemoved[currentIp] = null
+    } else {
+      toBeRemoved[ip] = null
+    }
+    setIpPool({ id, addresses: toBeRemoved })
+  }
+  _onChangeNewIps = (newIp, ipPoolId) =>
+    this.setState({ newIp: { ...this.state.newIp, [ipPoolId]: newIp } })
+
+  // Networks
+  _toggleNewNetworks = id => {
+    const { showNewNetworkForm } = this.state
+    this.setState({
+      showNewNetworkForm: { ...showNewNetworkForm, [id]: !(showNewNetworkForm && showNewNetworkForm[id]) }
+    })
+  }
+  _addNetworks = ({ id }) => {
+    setIpPool({
+      id,
+      networks: [ ...find(this.props.ipPools, ipPool => ipPool.id === id).networks, ...this.state.newNetworks[id] ]
+    })
+    this._toggleNewNetworks(id)
+  }
+  _deleteNetwork = ({ id, networks, networkId }) => {
+    const _networks = [ ...networks ]
+    const index = findIndex(_networks, network => network === networkId)
+    if (index !== -1) {
+      _networks.splice(index, 1)
+      setIpPool({
+        id,
+        networks: _networks
+      })
+    }
+  }
+  _onChangeNewNetworks = (newNetworks, ipPoolId) =>
+    this.setState({ newNetworks: { ...this.state.newNetworks, [ipPoolId]: map(newNetworks, network => network.id) } })
+  _getNetworkPredicate = ipPoolId => network =>
+    !includes(find(this.props.ipPools, ipPool => ipPool.id === ipPoolId).networks, network.id)
+
+  _ipColumns = () => [
+    {
+      name: _('ipsName'),
+      itemRenderer: ipPool => <Text onChange={name => setIpPool({ id: ipPool.id, name })} value={ipPool.name} />,
+      sortCriteria: ipPool => ipPool.name
+    },
+    {
+      name: _('ipsPoolIps'),
+      itemRenderer: ipPool => <Container>
+        <Row>
+          <Col mediumSize={6} offset={5}><strong>{_('ipsVifs')}</strong></Col>
+        </Row>
+        {ipPool.addresses && map(formatIps(keys(ipPool.addresses)), ip => {
+          if (isObject(ip)) {
+            return <Row>
+              <Col mediumSize={5}>
+                <strong>{ip.first} <Icon icon='arrow-right' /> {ip.last}</strong>
+              </Col>
+              <Col mediumSize={1} offset={6}>
+                <ActionRowButton
+                  handler={this._deleteIp}
+                  handlerParam={{ ...ipPool, ip }}
+                  icon='delete'
+                />
+              </Col>
+            </Row>
+          }
+          return <Row>
+            <Col mediumSize={5}>
+              <strong>{ip}</strong>
+            </Col>
+            <Col mediumSize={6}>{!isEmpty(ipPool.addresses[ip].vifs)
```

Comment:
Use intermediary variables to avoid such long identifiers `ipPool.addresses[ip].vifs`)!